securedrop_client/db.py

```diff
@@ -54,7 +56,11 @@ def collection(self) -> List:
         collection.extend(self.messages)
         collection.extend(self.files)
         collection.extend(self.replies)
-        collection.sort(key=lambda x: x.file_counter)
+        collection.extend(self.draftreplies)
+        # Sort first by the file_counter, then by timestamp (used only for draft replies).
```

Comment:
I wonder if there's a more high-level visible place we should mention how we use timestamps for saved drafts. I can't think of one other than in a docstring for source or in our client architecture doc. I was just hoping to find more information about why we use the timestamp somewhere. I did find your PR comment:

> this is the local timestamp that the reply was sent and this will be used for ordering local replies when there are multiple attempted replies in between conversation items from the server.

So we could use a local_file_counter instead of timestamp right? I don't feel strongly about this but maybe it makes it clearer that we don't actually care about time, we just care about order in which a reply was drafted so that we can display the drafts in the correct order in the client?

Or perhaps we'll want to show the timestamp next to the draft to help the journalist remember when they drafted it?

Reply:
hmm yeah good point - how about I add a description of the ordering situation here to the wiki architecture page? 

i.e. saying something like

"draft replies store:
* a `file_counter` which points to the `file_counter` of the previously sent item. this enables us to interleave the drafts with the items from the source conversation fetched from the server, which do not have timestamps associated with them.
* a `timestamp` which contains the timestamp the draft reply was saved locally: this is used to order drafts in the case where there are multiple drafts sent after a given reply (i.e. when `file_counter` is the same for multiple drafts)"

with an example

I actually did call the `DraftReply.file_counter` field `local_file_counter` field (😇) but then renamed it back to `file_counter` to simplify the `source.collection.sort` key. You're right that we _could_ ditch `timestamp` and have two fields `file_counter` and `local_file_counter`. imho I figure is slightly more useful to have the actual timestamp locally for if we ever do want to expose the draft timestamp to users (I could imagine that being useful).